Description: Show Dialog in tint2conf about unsupported saving
 This patch opens a dialog in the Properties view of tint2conf
 informing the user, that saving is not yet supported.
Author: Sebastian Reichel <sre@debian.org>
Bug-Debian: 646055
Last-Update: 2011-10-23

--- a/src/tint2conf/main.c.orig	2011-10-23 09:45:15.000000000 +0200
+++ b/src/tint2conf/main.c	2011-10-23 10:54:07.000000000 +0200
@@ -325,6 +325,24 @@
 	}
 }
 
+static void dialogResponse(GtkWidget *w, int response)
+{
+	GtkWidget *dialog = NULL, *label, *content_area;
+
+	if(response == GTK_RESPONSE_APPLY || response == GTK_RESPONSE_OK) {
+		dialog = gtk_dialog_new_with_buttons ("Saving not yet supported",
+				 (GtkWindow*) w, GTK_DIALOG_MODAL, GTK_STOCK_OK, GTK_RESPONSE_NONE, NULL);
+		content_area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
+		label = gtk_label_new("Saving not yet supported");
+		g_signal_connect(dialog, "response", G_CALLBACK (gtk_widget_destroy), dialog);
+		gtk_container_add(GTK_CONTAINER(content_area), label);
+		gtk_widget_show_all(dialog);
+	}
+
+	if(response == GTK_RESPONSE_CANCEL || response == GTK_RESPONSE_OK) {
+		gtk_widget_destroy(w);
+	}
+}
 
 static void menuProperties()
 {
@@ -351,6 +369,7 @@
 		//*/
 		g_free(file);
 		
+		g_signal_connect(prop, "response", G_CALLBACK (dialogResponse), prop);
 	}
 }
 
